ipatests/test_xmlrpc/xmlrpc_test.py

```diff
@@ -306,24 +304,20 @@ class Declarative(XMLRPC_test):
     cleanup_commands = tuple()
     tests = tuple()
 
-    @classmethod
-    def setup_class(cls):
-        super(Declarative, cls).setup_class()
-        for command in cls.cleanup_commands:
-            cls.cleanup(command)
-
-    @classmethod
-    def teardown_class(cls):
-        for command in cls.cleanup_commands:
-            cls.cleanup(command)
-        super(Declarative, cls).teardown_class()
+    @pytest.fixture(autouse=True, scope="class")
+    def declarative_setup(self, request, xmlrpc_setup):
```

Comment:
I find this confusing. It's an instance method that is used as a class scope fixture. Shouldn't the method be a class method, too?

Reply:
Me too, but `@classmethod` decorator is not supported in class scoped fixtures:
https://github.com/pytest-dev/pytest/issues/3778#issuecomment-479294091
https://github.com/pytest-dev/pytest/pull/3781

Reply:
Recommended workaround https://github.com/pytest-dev/pytest/issues/3778#issuecomment-411899446 is identical to applied in this PR:
```diff
- cls = type(self)
+ cls = request.cls
```